### PR TITLE
Sampling from prior then passing through flow

### PR DESCRIPTION
**Pull request by pdura7**
Probably the sampled gaussian posterior should be passed through the flow network and only then decoded.

### CODE REVIEW

File: dynamic_flow_vae.py [hunk 1]
```diff
@@ -279,15 +279,20 @@ def decode(self, z):
...
             size: number of samples to generate.
+            sigma: scaling factor for the prior variance
         Returns:
             generated samples.
         """
-        z = torch.randn(size, self.latent_dim).cuda()
+        sigma = torch.tensor(sigma).cuda()
+        mean = torch.zeros(size, self.latent_dim).cuda()
+        log_var = torch.ones(size, self.latent_dim).cuda()
+        log_var = sigma.log() * log_var
```
Comment (ghost) [inline dynamic_flow_vae.py:294]: Maybe Tensor.expand is more clean?

File: static_flow_vae.py [hunk 2]
```diff
@@ -336,15 +336,20 @@ def decode(self, z):
             z = self.decoder[i](z)
         return z
 
-    def sample(self, size):
+    def sample(self, size, sigma=1.0):
```
Comment (ghost) [inline static_flow_vae.py:339]: Why it is implemented three times? :O

File: dynamic_flow_vae.py (hunk 1 above)
Comment (Kanbe-Kotori) [reply dynamic_flow_vae.py:294]: > Maybe Tensor.expand is more clean?

torch.full() is better I think...

**Comment by Kanbe-Kotori** [comment]
I really wonder why this model can directly obtain samples by putting the standard normal distribution (torch.randn) into the decoder (without using flow), until I see this.. It seems that the author has made some mistakes..

**Pull request closed without merging**